mm2src/coins/eth.rs

```diff
@@ -7026,11 +7256,12 @@ pub fn pubkey_from_extended(extended_pubkey: &Secp256k1ExtendedPublicKey) -> Pub
     pubkey_uncompressed
 }
 
-fn extract_gas_limit_from_conf(coin_conf: &Json) -> Result<EthGasLimit, String> {
-    if coin_conf["gas_limit"].is_null() {
+fn extract_gas_limit_from_conf<T: ExtractGasLimit>(coin_conf: &Json) -> Result<T, String> {
+    let key = T::key();
+    if coin_conf[key].is_null() {
         Ok(Default::default())
     } else {
-        json::from_value(coin_conf["gas_limit"].clone()).map_err(|e| e.to_string())
+        json::from_value(coin_conf[key].clone()).map_err(|e| e.to_string())
```

Comment:
```rust
coin_conf[key]
    .as_ref()
    .map_or_else(
        || Ok(Default::default()),
        |value| json::from_value(value.clone()).map_err(|e| e.to_string())
    )
```

Reply:
thats really hard to read, also `Value` doesnt impl `as_ref` and `map_or_else` methods